### DISCUSSION TITLE
How to build Julia with GCC and for ppc32?

### DISCUSSION THREAD

**Discussion by barracuda156** [JuliaLang/julia#44684] [General]
I am interested to build Julia for PowerPC Darwin.
Since clang is broken for ppc, it has to be done with gcc (gcc11 and gfortran are available).
From what I see Julia tries to force usage of clang and seems to lack ppc and ppc64 ABI.
Is there a way to fix that?

**Comment by vchuravy** [comment]
We only support ppc64le-gnu, and that just barely. Supporting ppc32 will involve a lot of work, and the fact that clang is broken doesn't instill confidence since both clang and julia use llvm.

**Reply by barracuda156** [reply]
> We only support ppc64le-gnu, and that just barely. Supporting ppc32 will involve a lot of work, and the fact that clang is broken doesn't instill confidence since both clang and julia use llvm.

It is possible to use `gcc` as a compiler instead of `clang`+`llvm`? I could look into `ppc` support in Julia whether it can be fixed for Darwin, but fixing `clang`/`llvm` is out of question.

**Reply by vchuravy** [reply]
We use LLVM as a backend, so you could compile Julia with GCC, but the code Julia generates still uses LLVM

**Reply by barracuda156** [reply]
Thank you. That is upsetting, but what to do. There are little hopes in fixing LLVM, it is substantially broken and no one has time to deal with it.

**Reply by vchuravy** [reply]
Yeah from our side supporting gcc as a JIT backend would be an enormous undertaking that no-one has the appetite for.

**Comment by barracuda156** [comment]
> We only support ppc64le-gnu, and that just barely. Supporting ppc32 will involve a lot of work, and the fact that clang is broken doesn't instill confidence since both clang and julia use llvm.

Is it possible to support GCC build? Forget ppc32, I can try fixing that, but GCC it a must for that.

**Reply by giordano** [reply]
> Is it possible to support GCC build? Forget ppc32, I can try fixing that, but GCC it a must for that.

Valentin [already answered](https://github.com/JuliaLang/julia/discussions/44684#discussioncomment-3292184) that question:

> We use LLVM as a backend, so you could compile Julia with GCC, but the code Julia generates still uses LLVM

To reiterate once more: the julia executable itself can in general be compiled with GCC, that's what everybody does on Linux, but you also need a working LLVM for the architecture you want to use, because that's what julia will use as backend to generate code.
